cmd/restic/cmd_rewrite_integration_test.go

```diff
@@ -128,14 +127,37 @@ func testRewriteMetadata(t *testing.T, metadata snapshotMetadataArgs) {
 }
 
 func TestRewriteMetadata(t *testing.T) {
-	newHost := "new host"
+	newHost := "new-host" // hostname cannot contain spaces according to Linux !
 	newTime := "1999-01-01 11:11:11"
 
 	for _, metadata := range []snapshotMetadataArgs{
-		{Hostname: "", Time: newTime},
-		{Hostname: newHost, Time: ""},
+		//{Hostname: "", Time: newTime}, // hostname must be valid !
+		//{Hostname: newHost, Time: ""}, // empty time string now not allowed any more !
 		{Hostname: newHost, Time: newTime},
 	} {
 		testRewriteMetadata(t, metadata)
 	}
 }
+
+func TestRewriteSnaphotSummary(t *testing.T) {
+	env, cleanup := withTestEnvironment(t)
+	defer cleanup()
+	snap := createBasicRewriteRepo(t, env)
+
+	// rewrite --snapshot-summary <snap>
+	rtest.OK(t, runRewrite(context.TODO(), RewriteOptions{SnapshotSummary: true}, env.gopts, []string{snap.String()}))
+	// since we have attached SnapshotSummary to <snap>, the call to runRewrite is a no op
+	testListSnapshots(t, env.gopts, 1)
+
+	// get repo so we can access snapshot strcuture - this is a hackish.
```

Comment:
There's no easy way to get rid of the snapshot summary. The cleanest variant is probably to open the repository with an exclusive lock, then call `restic.LoadSnapshot()`, modify it, store it using `restic.SaveSnapshot()` , delete the old one via `repo.RemoveUnpacked()` and directly unlock the repository afterwards.